Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -469,10 +437,11 @@ static void OnConnected(int connectionId)
         {
             if (LogFilter.Debug) Debug.Log("Server accepted client:" + connectionId);
 
-            // connectionId needs to be > 0 because 0 is reserved for local player
-            if (connectionId <= 0)
+            // are we expecting a local connection, but a local connection was
+            // already set? then something is wrong!
+            if (pendingLocalConnection && localConnection != null)
```

Comment:
how do you know this connection is the local one?  this connection could be coming from anywhere.  Are you just hoping that the local connection wins the race?